R/paginate.R

```diff
@@ -0,0 +1,331 @@
+
+#' Pagination
+#'
+#' @inheritParams req_perform
+#' @param next_request A callback function that takes a two arguments (the
+#'   original request and the response) and returns:
+#'
+#'   * a new [request] to request the next page or
+#'   * `NULL` if there is no next page.
+#' @param page_size A parameter object that specifies how the page size is added
+#'   to the request.
+#' @param total A character that specifies the path where in the body the field
+#'   with the total number of elements is stored.
+#' @param next_url A character that specifies the path where in the body the field
+#'   with the next url of the next page is stored.
+#' @param offset A parameter object that specifies how the offset is added to
+#'   the request.
+#' @param token_field A parameter object that specifies how the next token is
+#'   added to the request.
+#' @param next_token_field A character that specifies the path where in the body
+#'   the field with the token of the next page is stored.
+#'
+#' @return A modified HTTP [request].
+#' @export
+#'
+#' @examples
+#' request("https://pokeapi.co/api/v2/pokemon") %>%
+#'   req_paginate_next_url(
+#'     "next",
+#'     page_size = in_query("limit", 150L),
+#'     total = "count"
+#'   )
+req_paginate <- function(req,
+                         next_request,
+                         page_size = NULL,
+                         total = NULL) {
```

Comment:
Probably the `total` interface isn't flexible enough. The response might contain a) the number of elements (what I assume in this implementation) or b) the number of pages.